### ISSUE TITLE
Reproduced results differ from reported ARI scores using provided data and code

### ISSUE THREAD

**Issue by FamingZhaoOHSU** [cmzuo11/stClinic#1]
Hi, 

Thanks for sharing the code and dataset!

I ran the provided stClinic code on the official DLPFC data (151673–151676). While the pipeline runs without errors, the clustering results and ARI scores I get are much lower than reported. For example, ARI scores in the paper are ~0.7, but mine are only ~0.2–0.25 (see attached figure).

![Image](https://github.com/user-attachments/assets/5889db66-f042-418e-bb81-8921069bcdd4)

I used:
	•	Python 3.8, R 4.4.3, PyTorch 2.4.0+cu121
	•	Code and data from this repo
	•	Ran training + mclust_R with default args

Could you advise if there are any missing steps or settings needed to reproduce the paper results?

Thanks!

**Comment by FamingZhaoOHSU** [comment]
By the way, here is my UMAP plot (right panel):

<img width="1000" alt="Image" src="https://github.com/user-attachments/assets/a9858078-e6d2-4faf-99d8-12b4a122c797" />

**Comment by WJPina** [comment]
Got same problems:

![Image](https://github.com/user-attachments/assets/7b76b97f-2f0d-4e37-a1f5-453e88325cdc)
 
I guess it might be affected by some parameters (such as rad_cutoff, n_top_genes, lr_integration)?
I used default values since I didn't find the exact values in the tutorias (DLPFC_Unsupervised_Integration.py).

**Comment by WJPina** [comment]
I also got problems in reproducing Fig.3a since the source of histological annotations of BAS1 and BAS2 are ambiguous and relevant metadata can not be found neither in  10x dataset (https://www.10xgenomics.com/datasets/) nor h5ad file provided in Github.

Could you please give more details about this ?
Thanks!

**Comment by FamingZhaoOHSU** [comment]
I think they may be annotated by the pathologist from this paper or other publication.

**Comment by WJPina** [comment]
> I think they may be annotated by the pathologist from this paper or other publication.

Yes, but it's hard to find the exact source or metadata : (

**Comment by Paul-YPH** [comment]
I’m seeing the exact same behavior. Even on standard benchmark datasets (like DLPFC), I’m only getting an ARI of around 0.2, which is way off from the results in the paper.

And I got:

<img width="149" height="161" alt="Image" src="https://github.com/user-attachments/assets/44ce09ad-b0ef-426e-b00c-3d1a40e42b9e" />

The learning rate was adjusted because the default setting resulted in errors; PCs/KNN represents the optimal parameters I derived through ARI optimization, while keeping all other conditions constant.

Here’s the simplified logic I’m running:
```
for ann in ann_list:
    stClinic.Cal_Spatial_Net(ann, k_cutoff=10, model='KNN')
    # Using seurat_v3 as per official examples
    sc.pp.highly_variable_genes(ann, flavor="seurat_v3", n_top_genes=3000)
    sc.pp.normalize_total(ann, target_sum=1e4)
    sc.pp.log1p(ann)
    ann = ann[:, ann.var['highly_variable']]
    sc.tl.pca(ann, n_comps=200)
adata = stClinic.train_stClinic_model(adata, n_centroids=n, lr=0.00005, device=device)

**Comment by JAAY-AI** [comment]
Could the author please respond to these questions?
